src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java

```diff
@@ -58,26 +92,48 @@ public DriveBaseSubsystem(Joystick joystick) {
     m_motorControllers[Constants.driveLeftRearIndex].setFollow(m_motorControllers[Constants.driveLeftFrontIndex]);
     m_motorControllers[Constants.driveRightRearIndex].setFollow(m_motorControllers[Constants.driveRightFrontIndex]);
 
+    if (Robot.isSimulation()) {
+      m_DifferentialDrivetrainSim = new DifferentialDrivetrainSim(
+        DCMotor.getNEO(2),       // 2 NEO motors on each side of the drivetrain.
+        7.29,                    // 7.29:1 gearing reduction.
+        7.5,                     // MOI of 7.5 kg m^2 (from CAD model).
+        60.0,                    // mass of the robot
+        Units.inchesToMeters(3), // The robot wheel radius
+        0.7112,                  // The track width
+        VecBuilder.fill(0.001, 0.001, 0.001, 0.1, 0.1, 0.005, 0.005)
+      );
+
+        // standard deviations for measurement noise: x and y: 0.001m heading: 0.001 rad  l and r velocity: 0.1m/s  l and r position: 0.005m
+    }
     // differential drive
     m_differentialDrive = new DifferentialDrive(m_motorControllers[Constants.driveLeftFrontIndex].getSparkMax(), 
                                           m_motorControllers[Constants.driveRightFrontIndex].getSparkMax());
   }
 
   @Override
   public void periodic() {
-    arcadeDrive();  // periodically runs the arcadeDrive function, avoid scheduling the command
 
     // Update the smart dashboard in here, runs a for loop so it does it for every motor
     for(int i = 0; i < m_motorControllers.length; i++) {
       m_motorControllers[i].updateSmartDashboard();
+      m_odometry.update(m_gyro.getRotation2d(),
+                    m_leftEncoder.getDistance(),
+                    m_rightEncoder.getDistance());
+      m_field.setRobotPose(m_odometry.getPoseMeters());
```

Comment:
Move the setRobotPose to simulationPeriodic. At least for now. It might be usable in normal operation, but not currently

Reply:
done